src/document/utils/fill-builder.js

```diff
@@ -0,0 +1,30 @@
+import { fill } from "../builders.js";
+
+function fillBuilder() {
+  const parts = [];
+  let pendingDocs = [];
+  function flush() {
+    parts.push(pendingDocs);
+    pendingDocs = [];
+  }
+
+  const self = {
+    append(doc) {
+      pendingDocs.push(doc);
+      return self;
+    },
+    appendLine(doc) {
+      flush();
+      pendingDocs = [];
+      parts.push(doc);
+      return self;
+    },
+    build() {
+      flush();
+      return fill(parts);
+    },
+  };
+  return self;
+}
```

Comment:
**Possible alternative API**
```js
const result = buildFill(($) => {
  $.append("a");
  $.append("b");
  $.appendLine(line);
  $.append("c");
})
```
instead of
```js
const builder = fillBuilder();
builder.append("a");
builder.append("b");
builder.appendLine(line);
builder.append("c");
const result = builder.build();
```